tools/ak2-core.sh

```diff
@@ -338,6 +339,9 @@ write_boot() {
 # backup_file <file>
 backup_file() { test ! -f $1~ && cp $1 $1~; }
 
+# restore_file <file>
+restore_file() { test -f $1~ && mv -f $1~ $1; }
```

Comment:
Huh, I guess I just realized this is kind of awkward because `restore_file` would always need to be used before `backup_file`. @osm0sis maybe it would be smart to add this to `backup_file`? What is the use case for `backup_file` without a way to restore lol?

Reply:
It was always just to have both files present in a developer's testing phase of AK2 to diff and see that the intended changes were made. Actual backup and restore introduces too many variables, the most important being that you wouldn't want to restore an old original file over an updated one from a newer ROM boot.img.

Reply:
That said, I guess with the caveats that if it's used for a ramdisk-only flash, and is in a separate "restore" zip to undo whatever was done by the main patching zip, it could make sense. You just wouldn't want to have it to `restore_file blah; backup_file blah;` as part of a kernel flash or the patching zip since thar be dragons.

I'll add it if it's separated from these other unnecessary changes, and we'll have to update the documentation to properly explain the use of backup and restore.